x-pack/plugins/ml/public/embeddables/common/anomaly_detection_embeddable.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  Embeddable,
+  type EmbeddableInput,
+  type EmbeddableOutput,
+  type IContainer,
+} from '@kbn/embeddable-plugin/public';
+import { type DataView } from '@kbn/data-views-plugin/common';
+import { type DataViewsContract } from '@kbn/data-views-plugin/public';
+import { firstValueFrom } from 'rxjs';
+import { type AnomalyDetectorService } from '../../application/services/anomaly_detector_service';
+
+export type CommonInput = { jobIds: string[] } & EmbeddableInput;
+
+export type CommonOutput = { indexPatterns?: DataView[] } & EmbeddableOutput;
+
+export abstract class AnomalyDetectionEmbeddable<
+  Input extends CommonInput,
+  Output extends CommonOutput
+> extends Embeddable<Input, Output> {
+  // Need to defer embeddable load in order to resolve data views
+  deferEmbeddableLoad = true;
+
+  protected constructor(
+    initialInput: Input,
+    private anomalyDetectorService: AnomalyDetectorService,
+    private dataViewsService: DataViewsContract,
+    parent?: IContainer
+  ) {
+    super(
+      initialInput,
+      {
+        defaultTitle: initialInput.title,
+        defaultDescription: initialInput.description,
+      } as Output,
+      parent
+    );
+
+    this.initializeOutput(initialInput).finally(() => {
+      this.setInitializationFinished();
+    });
+  }
+
+  protected async initializeOutput(initialInput: CommonInput) {
+    const { jobIds } = initialInput;
+
+    try {
+      const jobs = await firstValueFrom(this.anomalyDetectorService.getJobs$(jobIds));
+
+      // First get list of unique indices from the selected jobs
+      const indices = new Set(jobs.map((j) => j.datafeed_config!.indices).flat());
```

Comment:
Is there anything we can do to support filtering where the job has been created from a comma-separated list of index patterns?

For example with a job created from a data view `gallery-2021-11,gallery-2021-12` auto-suggest doesn't work.

Reply:
updated in https://github.com/elastic/kibana/pull/164485/commits/ba6599dc2756e41b8d8087b5af40dc2d6933df02